portal/src/main/webapp/js/src/cbio-util.js

```diff
@@ -339,15 +339,27 @@ cbio.util = (function() {
     };
 
     /**
-     * Replaces problematic characters with an underscore for the given string.
-     * Those characters cause problems with the properties of an HTML object,
-     * especially for the id and class properties.
+     * Gives a unique id for each given string. All strings are saved in a map,
+     * so if the same string is called more than once, it returns the unique id
+     * previously assigned. This unique id does not contain characters that cause
+     * problems with the properties of an HTML object, especially for the id and
+     * class properties.
      *
      * @param property  string to be modified
      * @return {string} safe version of the given string
      */
     var safeProperty = function(property) {
-        return property.replace(/[^a-zA-Z0-9-]/g, '_');
+        return uidProperty(property);
+    };
+
+    var uidPropertyMap = {};
+    var uidCount = 0;
+    var uidProperty = function(property) {
+        if (!uidPropertyMap[property]) {
+            uidPropertyMap[property] = "uid"+uidCount;
```

Comment:
Maybe the description needs to be revised. Gene set names contain characters like dots, which cause problems if they are used as ids for HTML elements. By using the original function, the problematic characters are replaced, but this does not necessarily make them unique (for example, "GSVA/1" and "GSVA.1" would both be transformed to "GSVA_1").  Therefore, we decided to change the function to explicitly make a new unique id each time the function is called.